File: 백준/Silver/11047. 동전 0/동전 0.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int N = Integer.parseInt(input[0]);
        long K = Long.parseLong(input[1]);
        long[] coins = new long[N];
        int idx = 0;

        for (int i = 0; i < coins.length; i++) {
            coins[i] = Integer.parseInt(br.readLine());

            if (coins[i] <= K) {
                idx = i;
            }
        }

        int cnt = 0;

        for (int i = idx; i >= 0; i--) {
            cnt += (int) (K / coins[i]);
            K %= coins[i];

            if (K == 0) {
                break;
            }
        }

        System.out.println(cnt);
    }
}

File: 백준/Silver/11047. 동전 0/동전 0.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

let [N, ...coins] = input;
[N, K] = N.toString().split(" ").map(Number);
coins = coins.map(Number).sort((a, b) => b - a);

let cnt = 0;

for (const coin of coins) {
  cnt += parseInt(K / coin);
  K %= coin;

  if (K === 0) {
    break;
  }
}

console.log(cnt);
